pkg/ccl/sqlproxyccl/balancer/conn_tracker.go

```diff
@@ -134,59 +143,129 @@ func logTrackerEvent(event string, sa *ServerAssignment) {
 	log.Infof(logCtx, "%s: %s", event, sa.Addr())
 }
 
-// tenantEntry is a connection tracker entry that stores connection information
-// for a single tenant. All methods on tenantEntry are thread-safe.
-type tenantEntry struct {
-	// mu synchronizes access to the list of connections associated with this
-	// tenant.
-	mu struct {
-		syncutil.Mutex
+// refreshPartitionsLoop runs on a background goroutine to continuously refresh
+// partitions of assignments in the tracker.
+func (t *ConnTracker) refreshPartitionsLoop(ctx context.Context) {
+	timer := t.timeSource.NewTimer()
+	defer timer.Stop()
+	for {
+		timer.Reset(partitionsRefreshInterval)
+		select {
+		case <-ctx.Done():
+			return
+		case <-timer.Ch():
+			timer.MarkRead()
+			t.refreshPartitions()
+		}
+	}
+}
 
-		// assignments refers to a set of ServerAssignment objects.
-		//
-		// TODO(jaylim-crl): Break this into active and idle partitions.
-		assignments map[*ServerAssignment]struct{}
+// refreshPartitions refreshes the partitions of all assignments for all tenants
+// associated with the tracker.
+func (t *ConnTracker) refreshPartitions() {
+	tenantIDs := t.getTenantIDs()
+	for _, tenantID := range tenantIDs {
+		e := t.getEntry(tenantID, false /* allowCreate */)
+		if e != nil {
+			e.refreshPartitions()
+		}
+	}
+}
+
+// tenantEntry is a tracker entry that stores server assignments for a single
+// tenant. All methods on tenantEntry are thread-safe.
+//
+// There are two partitions for server assignments: active and idle. New
+// assignments are placed in the active partition. A background goroutine in the
+// tracker will attempt to move assignments around based on the state of the
+// assignment's owner (i.e. through the owner's IsIdle method).
+type tenantEntry struct {
+	mu          syncutil.Mutex
+	assignments struct {
+		active map[*ServerAssignment]struct{}
```

Comment:
nit: I like to represent sets in go as map[key]bool. It makes it easier to check for the presence of an item in the set. 

E.g.
```
if tenantEntry.assignments.active[sa] {
     // server assignment is active
}
```